In [1]:
import numpy as np

In [3]:
x = np.array([3,4,5])
y = np.array([4,9,7])
# treats them as matrices
print(x + y)

[ 7 13 12]


In [4]:
x = np.array([[1,2],[3,4]])
# follows the structure
print(x)

[[1 2]
 [3 4]]


In [5]:
# shows the number of dimensions
print(x.ndim)

2


In [6]:
# shows the type
print(x.dtype)
z = np.array([[1 , 2],[42,53],[54,2]])
print(z.dtype)

int64
int64


In [7]:
# shows the number of rows and columns
print(z.shape)

(3, 2)


In [8]:
# dont need to use print here
z

array([[ 1,  2],
       [42, 53],
       [54,  2]])

In [9]:
# reshapes the data as long as it fits
print(z.reshape(2,3))
print()
print(z.reshape(1,6))

[[ 1  2 42]
 [53 54  2]]

[[ 1  2 42 53 54  2]]


In [10]:
# swaps the rows and cols aka transpose
print(z)
print()
print(z.T)

[[ 1  2]
 [42 53]
 [54  2]]

[[ 1 42 54]
 [ 2 53  2]]


In [ ]:
# generates the square root of the entries
np.sqrt(z)
# or x**0.5

In [ ]:
# generates the square of the elements
z**2

In [ ]:
# generates a 1 row matrix with 50 datapoints
x = np.random.normal(size = 50)
x

In [ ]:
# if given loc, it generates noise from that loc 50
y = x + np.random.normal(loc=50,scale=1, size=50)
y

the off-diagnol element shows and computes the correlation coefficient matrix between x and y.B
it detects relationships either the 2 act together or in opposite
if one is 1,2,3,4 and the other is 10,13,15,17 the both rise so its 1
if the other is 39,20,12,5 then they are at perfect opposite of each other resulting in -1

In [ ]:
np.corrcoef(x,y)

In [ ]:
rng = np.random.default_rng(1303)
print(rng.normal(scale=5, size=2))


[ 4.09482632 -1.07485605]


In [33]:
# np.mean() np.var() and np.std() give out mean, variance and standard deviation of arrays
rng = np.random.default_rng(1303)
y = rng.standard_normal(10)
y = np.array([[1,2,3],[4,5,6]])
print(y)
print('mean', y.mean())
print('var', y.var())
print('standard deviation', y.std())

[[1 2 3]
 [4 5 6]]
mean 3.5
var 2.9166666666666665
standard deviation 1.707825127659933


Mean:
    Definition: The arithmetic mean, calculated as the sum of all values divided by the number of values.
    Example: For [1,2,3,4,5], mean=(1+2+3+4+5)/5=3mean=(1+2+3+4+5)/5=3.
    Type: A specific type of average.

Median:
    Definition: The middle value when data is sorted. For even-sized datasets, it’s the average of the two middle values.
    Example: For [1,3,5,7,9], median=5median=5. For [1,2,3,4,5,6], median=(3+4)/2=3.5median=(3+4)/2=3.5

Mode:
    Definition: The most frequently occurring value(s) in the dataset. If all values occur equally, there’s no mode.
    Example: For [1,2,2,3,4], mode=2mode=2. For [1,2,3,4], there’s no mode.

Average:
    Definition: A general term for a central value of a dataset. It can refer to mean, median, or mode (context-dependent).
    Example: In most casual usage, average = mean.